File: Leetcode/1151. Minimum Swaps to Group All 1's Together.cpp
// https://leetcode.com/problems/minimum-swaps-to-group-all-1s-together/

class Solution {
public:
    int minSwaps(vector<int>& data) {
		int result = data.size();
		int oneCount = 0;
		for (int i = 0; i < data.size(); i++) {
			if (data.at(i) == 1) {
				oneCount++;
			}
		}
		if (oneCount == 0) {
			return 0;
		}
		int windowCount = 0;
		for (int i = 0; i < oneCount - 1; i++) {
			if (data.at(i) == 1) {
				windowCount++;
			}
		}
		for (int i = 0; i < data.size() - oneCount + 1; i++) {
			windowCount += data.at(i + oneCount - 1);
			result = min(result, oneCount - windowCount);
			windowCount -= data.at(i);
		}
		return result;
    }
};

File: Leetcode/1151. Minimum Swaps to Group All 1's Together.py
# https://leetcode.com/problems/minimum-swaps-to-group-all-1s-together/

from typing import List


class Solution:
    def minSwaps(self, data: List[int]) -> int:
        result = len(data)
        oneCount = data.count(1)
        if oneCount == 0:
            return 0
        windowCnt = sum(data[:oneCount - 1])
        for i in range(len(data) - oneCount + 1):
            windowCnt += data[i + oneCount - 1]
            result = min(result, oneCount - windowCnt)
            windowCnt -= data[i]
        return result
